x/dsmr/p2p.go

```diff
@@ -135,6 +135,12 @@ func (c ChunkSignatureRequestVerifier[T]) Verify(
 			Message: err.Error(),
 		}
 	}
+	if err := c.storage.CheckRateLimit(chunk); err != nil {
+		return &common.AppError{
+			Code:    p2p.ErrUnexpected.Code,
+			Message: err.Error(),
+		}
+	}
```

Comment:
Can we add this above `c.storage.GetChunkBytes(...)` ?

It's preferred to perform the lightest checks first, so that if a request will fail, we minimize the resources used to fail it. This is not a critical difference since a malicious peer can inspect our code and craft a response that will fail at a specific check, but it's a good practice.

Reply:
(unrelated to this PR) I think the above check if the bytes are present is unnecessary.

Since we ignore the result of the query, the only case where it changes our handling is if we hit an unexpected database error (ie. not `database.ErrNotFound`). I think we're just wasting a potential database read here.